### ISSUE TITLE
Support a REST API: Add Workbook Item Description

### ISSUE THREAD

**Issue by allturnip** [tableau/server-client-python#1147]
## Summary
Provide a setter for the workbook item Description Field

## Request Type
Support a REST API

The API: PUT /api/api-version/sites/site-id/workbooks/workbook-id supports a description field. 
the Schema definition for WorkbookType contains an element xs:attribute name="description" type="xs:string"

The Python API currently does not support this element.
It could be added by modifying:
1. tableauserverclient/models/workbook_item.py -- adding a setter for description
2. tableauserverclient/server/request_factory.py:
    update class WorkbookRequest : _generate_xml : in the same method as other elements, add a description :  
        if workbook_item.description:
              workbook_element.attrib['description'] = workbook_item.description
   
 add same code to procedure update_req.

This change allows a workbook description to be added at time of upload, preventing manual intervention after report promotions, migrations, or other automated processes.

**Comment by nosnilmot** [comment]
The current REST API [Workbooks Methods](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_workbooks_and_views.htm#update_workbook) documentation does not include a `description` field in any publish or update methods, only in responses

**Comment by jacalata** [comment]
That's correct: the description field is not settable through the API currently. I don't think there's a real reason for that, so I'll prod the team that owns that area.

**Comment by geerties99** [comment]
Has the setter for description been added already? 
Could need it for a current project.

**Comment by AverageJoeGuy-tech** [comment]
Has the status of this changed? The REST API currently allows workbook description to be set/changed in PUT.

**Comment by jacalata** [comment]
Closing — this was implemented across a few PRs:

- PR #533 added `description` as a read-only attribute on `WorkbookItem`
- Later commit `0ddae7c` (feat: add description support on wb publish) added support for setting the description on publish
- PR #1488 added the docstring for the attribute

Set it on the WorkbookItem before publish or before calling update.
